src/lib/modernExtend.ts

```diff
@@ -151,108 +374,274 @@ export function onOff(args?: OnOffArgs): ModernExtend {
-                    if (property.attribute === 'currentSummDelivered') change = [0, change];
-                    items.push({attribute: property.attribute, min: '10_SECONDS', max: 'MAX', change});
+    if (args.bind) {
+        result.configure = async (device, coordinatorEndpoint, logger) => {
+            if (args.endpointNames) {
+                const endpointsMap = new Map<string, boolean>(args.endpointNames.map((e) => [e, true]));
+                const endpoints = device.endpoints.filter((e) => endpointsMap.has(e.ID.toString()));
+                for (const endpoint of endpoints) {
+                    await endpoint.bind('genOnOff', coordinatorEndpoint);
                 }
-                if (items.length) {
-                    await setupAttributes(endpoint, coordinatorEndpoint, cluster, items, logger);
+            } else {
+                const endpoints = getEndpointsWithOutputCluster(device, 'genOnOff');
+                for (const endpoint of endpoints) {
+                    await endpoint.bind('genOnOff', coordinatorEndpoint);
                 }
             }
-        }
-    };
+        };
+    }
 
-    return {exposes, fromZigbee, toZigbee, configure, isModernExtend: true};
+    return result;
 }
 
-export interface LightArgs {
-    effect?: boolean, powerOnBehavior?: boolean, colorTemp?: {startup?: boolean, range: Range},
+export function customTimeResponse(start: '1970_UTC' | '2000_LOCAL'): ModernExtend {
+    // The Zigbee Cluster Library specification states that the genTime.time response should be the
+    // number of seconds since 1st Jan 2000 00:00:00 UTC. This extend modifies that:
+    // 1970_UTC: number of seconds since the Unix Epoch (1st Jan 1970 00:00:00 UTC)
+    // 2000_LOCAL: seconds since 1 January in the local time zone.
+    // Disable the responses of zigbee-herdsman and respond here instead.
+    const onEvent: OnEvent = async (type, data, device, options, state: KeyValue) => {
+        if (!device.customReadResponse) {
+            device.customReadResponse = (frame, endpoint) => {
+                if (frame.isCluster('genTime')) {
+                    const payload: KeyValue = {};
+                    if (start === '1970_UTC') {
+                        const time = Math.round(((new Date()).getTime()) / 1000);
+                        payload.time = time;
+                        payload.localTime = time - (new Date()).getTimezoneOffset() * 60;
+                    } else if (start === '2000_LOCAL') {
+                        const oneJanuary2000 = new Date('January 01, 2000 00:00:00 UTC+00:00').getTime();
+                        const secondsUTC = Math.round(((new Date()).getTime() - oneJanuary2000) / 1000);
+                        payload.time = secondsUTC - (new Date()).getTimezoneOffset() * 60;
+                    }
+                    data.endpoint.readResponse('genTime', data.meta.zclTransactionSequenceNumber, payload).catch((e) => {
```

Comment:
This reverts my fix: https://github.com/Koenkk/zigbee-herdsman-converters/commit/4d0dc21f642833cec196d4082bcb3b3f287a81ca

Reply:
Fixed